src/client/datascience/jupyter/debuggerVariables.ts

```diff
@@ -245,24 +245,21 @@ export class DebuggerVariables extends DebugLocationTracker
                 this.currentVariablesReference = newVariablesReference;
                 this.currentSeqNumsForVariables.clear();
             }
-        } else if (
-            message.type === 'response' &&
-            message.command === 'variables' &&
-            message.body &&
-            this.currentSeqNumsForVariables.has(message.request_seq)
-        ) {
-            // If using the interactive debugger, update our variables.
-            // eslint-disable-next-line
-            // TODO: Figure out what resource to use
-
-            // Only update variables if it came from a "scopes" command and not a "hover"
-            // 1. Scopes command will come first with a variablesReference number
-            // 2. onWillReceiveMessage will have that variablesReference and
-            // will request for variables with a seq number
-            // 3. We only updateVariables if the seq number is one of the sequence numbers that
-            // came with the most recent 'scopes' variablesReference
-
-            this.updateVariables(undefined, message as DebugProtocol.VariablesResponse);
+        } else if (message.type === 'response' && message.command === 'variables' && message.body) {
+            if (this.currentSeqNumsForVariables.has(message.request_seq)) {
+                // If using the interactive debugger, update our variables.
+                // TODO: Figure out what resource to use
+
+                // Only update variables if it came from a "scopes" command and not a "hover"
+                // 1. Scopes command will come first with a variablesReference number
+                // 2. onWillReceiveMessage will have that variablesReference and
+                // will request for variables with a seq number
+                // 3. We only updateVariables if the seq number is one of the sequence numbers that
+                // came with the most recent 'scopes' variablesReference
+                this.updateVariables(undefined, message as DebugProtocol.VariablesResponse);
```

Comment:
Looks like this regressed when a fix was made to not update variables on hover requests. The monkey patching can happen for any sequence number (such as fetching child variables from variables view). All it's doing is checking the type to see if it's a variable supported by data viewer, safe to do for anything.